src/lwc/rd2PauseForm/rd2PauseForm.js

```diff
@@ -0,0 +1,421 @@
+import { LightningElement, api, track, wire } from 'lwc';
+import { showToast, constructErrorMessage, isNull } from 'c/utilCommon';
+import { getRecord } from 'lightning/uiRecordApi';
+
+import NAME_FIELD from '@salesforce/schema/npe03__Recurring_Donation__c.Name';
+
+import header from '@salesforce/label/c.RD2_PauseHeader';
+import description from '@salesforce/label/c.RD2_PauseDescription';
+import loadingMessage from '@salesforce/label/c.labelMessageLoading';
+import cancelButton from '@salesforce/label/c.stgBtnCancel';
+import saveButton from '@salesforce/label/c.stgBtnSave';
+import okButton from '@salesforce/label/c.stgLabelOK';
+import selectedRowsSummaryPlural from '@salesforce/label/c.RD2_PauseSelectedInstallmentTextPlural';
+import selectedRowsSummarySingular from '@salesforce/label/c.RD2_PauseSelectedInstallmentTextSingular';
+import saveSuccessMessage from '@salesforce/label/c.RD2_PauseSaveSuccessMessage';
+import deactivationSuccessMessage from '@salesforce/label/c.RD2_PauseDeactivationSuccessMessage';
+import rdClosedMessage from '@salesforce/label/c.RD2_PauseClosedRDErrorMessage';
+import permissionRequired from '@salesforce/label/c.RD2_PausePermissionRequired';
+import insufficientPermissions from '@salesforce/label/c.lblInsufficientPermissions';
+
+import getPauseData from '@salesforce/apex/RD2_PauseForm_CTRL.getPauseData';
+import getInstallments from '@salesforce/apex/RD2_PauseForm_CTRL.getInstallments';
+import savePause from '@salesforce/apex/RD2_PauseForm_CTRL.savePause';
+
+export default class Rd2PauseForm extends LightningElement {
+
+    labels = Object.freeze({
+        header,
+        description,
+        loadingMessage,
+        cancelButton,
+        saveButton,
+        okButton,
+        selectedRowsSummaryPlural,
+        selectedRowsSummarySingular,
+        saveSuccessMessage,
+        deactivationSuccessMessage,
+        rdClosedMessage,
+        permissionRequired,
+        insufficientPermissions
+    });
+
+    @api recordId;
+    recordName;
+
+    @track isLoading = true;
+    @track hasAccess = true;
+    @track isRDClosed;
+    @track isSaveDisplayed;
+    @track isSaveDisabled = false;
+    @track pageHeader = '';
+    @track pausedReason = {};
+    scheduleId;
+
+    maxRowDisplay = 12;
+    maxRowSelection = 12;
+    selectedIds = [];
+    @track selectedRowsSummary = null;
+    @track columns = [];
+    @track installments;
+
+    @track error = {};
+
+    /***
+    * @description 
+    */
+    connectedCallback() {
+        this.init();
+    }
+
+    /***
+    * @description Group various calls to Apex
+    */
+    init = async () => {
```

Comment:
I'm intrigued by the use of `async()` in some of the functionality here. How is this used to improve page functionality and/or is it worth writing up a document on when/why we should use `async` in LWC controllers?

Reply:
I did not measure response time, but when not using `async()`, the form took a bit longer to load. Also, waiting for the PauseData ensures the page rendering completes and is useable (otherwise we need to have flags for the load complete (form ready)). The answer to this question and `data-id` vs `data-qa-locator` might require team discussion. I can start discussion offline. In meantime, the form works in various scenarios and the xpath can be determined by the automation, and if there is any change (which I doubt) based on the discussion, we can change it before merging complete Pause functionality into the `master`.

Reply:
Thanks for clarifying. I'm effectively wondering when we should consider using `async` and when we shouldn't.